include/world_builder/features/subducting_plate_models/temperature/mass_conserving.h

```diff
@@ -0,0 +1,117 @@
+/*
+  Copyright (C) 2018 - 2021 by the authors of the World Builder code.
+
+  This file is part of the World Builder.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef WORLD_BUILDER_FEATURES_SUBDUCTING_PLATE_MODELS_TEMPERATURE_MASS_CONSERVING_H
+#define WORLD_BUILDER_FEATURES_SUBDUCTING_PLATE_MODELS_TEMPERATURE_MASS_CONSERVING_H
+
+
+#include "world_builder/features/subducting_plate_models/temperature/interface.h"
+#include "world_builder/features/utilities.h"
+
+
+namespace WorldBuilder
+{
+
+  namespace Features
+  {
+    namespace SubductingPlateModels
+    {
+      namespace Temperature
+      {
+        /**
+         * This class represents a subducting plate temperature model. The temperature
+         * model uses the heat content (proportional to to thermal mass anomaly) to
+         * define a smooth temperature profile that conserves mass along the slab length.
+         * An empirical (linear) model is used to define how the minimum temperature
+         * increases with depth and how the location of the minimum temperature shifts
+         * into the slab interior. The slab is divided in to top and bottom parts,
+         * which meet at the location where the minimum temperature occurs in the slab.
+         * For the bottom slab the temperature is defined by a half-space cooling model.
+         * For the top of the slab the temperature is defined by one side of a 1D infinite
+         * space cooling model. The age of the overriding plate is used so the slab temperature
+         * at shallow depth smoothly transitions to the temperature of the overriding plate:
+         * this is not perfect, and is affected by the value of "top truncation" parameter
+         * subducting plate. Also note that the parameter "thickness" for the subducting plate_velocity
+         * needs to be defined but is not used.
```

Comment:
Could you add a note here warning about the internal parameters which are going to change in the future.